shell/renderer/pepper_helper.cc

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "shell/renderer/pepper_helper.h"
+
+#include <memory>
+
+#include "chrome/renderer/pepper/chrome_renderer_pepper_host_factory.h"
+#include "chrome/renderer/pepper/pepper_shared_memory_message_filter.h"
+#include "content/public/renderer/renderer_ppapi_host.h"
+#include "electron/shell/renderer/electron_renderer_pepper_host_factory.h"
+#include "ppapi/host/ppapi_host.h"
+
+PepperHelper::PepperHelper(content::RenderFrame* render_frame)
+    : RenderFrameObserver(render_frame) {}
+
+PepperHelper::~PepperHelper() {}
+
+void PepperHelper::DidCreatePepperPlugin(content::RendererPpapiHost* host) {
+  // TODO(brettw) figure out how to hook up the host factory. It needs some
+  // kind of filter-like system to allow dynamic additions.
+  host->GetPpapiHost()->AddHostFactoryFilter(
+      std::make_unique<ChromeRendererPepperHostFactory>(host));
+  host->GetPpapiHost()->AddHostFactoryFilter(
+      std::make_unique<ElectronRendererPepperHostFactory>(host));
```

Comment:
What is `ElectronRendererPepperHostFactory` gonna handle that is not handled by the chrome couterpart ?

Reply:
UMA blackholing

Reply:
We already do that in `pepper_plugin_support.patch`, is the plan to reduce the patch surface ?

Reply:
that patch doesn't blackhole UMA, it just doesn't create the UMA resource host at all. That ends up resulting in a DCHECK() when pepper plugins try to load the UMA interface (which the PDF plugin does). The UMA host in chrome relies on a bunch of chrome-specific stuff which I wanted to avoid, so building our own stub version seemed like the best way to go. An alternative path would be to patch the PDF plugin to not call any UMA hooks.

Reply:
Thanks for the context! in that case, this is fine as it is :)